CefSharp/Internals/PerBrowserJavaScriptObjectCache.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2023 The CefSharp Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+using System;
+using System.Collections.Generic;
+
+namespace CefSharp.Internals
+{
+    /// <summary>
+    /// Render Process JavaScript Binding (JSB) object cache
+    /// Stores bound objects per CefBrowser.
+    /// </summary>
+    public class PerBrowserJavaScriptObjectCache : IJavaScriptObjectCache
+    {
+        private readonly Dictionary<int, Dictionary<string, JavascriptObject>> cache
+            = new Dictionary<int, Dictionary<string, JavascriptObject>>();
+
+        /// <inheritdoc/>
+        public void ClearCache(int browserId)
+        {
+            cache.Remove(browserId);
+        }
+
+        /// <inheritdoc/>
+        public void InsertOrUpdate(int browserId, IList<JavascriptObject> javascriptObjects)
+        {
+            var dict = GetCacheInternal(browserId);
+
+            foreach (var obj in javascriptObjects)
+            {
+                if (dict.ContainsKey(obj.Name))
+                {
+                    dict.Remove(obj.Name);
+                }
+
+                dict.Add(obj.Name, obj);
```

Comment:
Nit: could be simplified to:
```suggestion
                dict[obj.Name] = obj;
```